### ISSUE TITLE
Change ownership

### ISSUE THREAD

**Issue by maxolasersquad** [linappleii/linapple#4]
Before I go moving this to the [linappleii](https://github.com/orgs/linappleii/people) namespace, I just want to open a conversation up about the move. I'm wanting to confirm that @dabonetn and @iKarith are wanting to take over this project.

**Comment by iKarith** [comment]
I've invited @ghedger and @cpressey to join that project namespace as well.  From discussion in RasppleII/a2cloud#37, it seems both are interested in seeing someone pick this up and begin collecting patches, but neither of them want to be responsible for it in the end.  Likewise, I don't think @dabonetn does, he's got a goal in mind for the end result.

So I guess it falls to me.  That's fine, I'll do it until someone else comes along with more time and more skill to curate the thing.

I've invited Greg and Chris both, and you're welcome too.  If anyone sees something that needs doing and they're the person to do it, have at it.  If not, no worries.  :)  This codebase has sat mostly untouched with a few flurries of fixes and small additions since I think like 2005 or so?

Bug fixes, go ahead and commit.  Bigger changes, do it with a PR so people have a chance to comment before it goes in.  How long … I guess it'll depend how responsive people are.  If people start getting involved who have more forceful personalities we may need to look at more structure, but as long as it's just a few guys who are too busy to do much except in brief spurts who are basically trying to accomplish the same basic goals, I think we'll be fine.

**Comment by ghedger** [comment]
I'll have a look when it's up.

For organizational purposes, I think it would be wise to keep the newer and more conventional build structure in place instead of having the files (source and headers, obj, executable) all in the same directory as it was before, and keep it moving forward.  That way, there's no need to change the Makefile when adding a new source file if that becomes necessary.

From a merge strategy standpoint though, if this creates undue chaos due to file mapping I'm fine keeping the old build structure provided it's temporary and we can be again reorganize it once the bulk of the churn is finished.

**Comment by iKarith** [comment]
For the merge, I'm pulling in Chris's stuff first, then I'll pull yours in on top of it, that way the re-org happens after all the hard work of merging is already done.  ;)  I'd have made different choices for the re-org personally, but I agree that some changes needed making with the build.  I had a Makefile patch against 2b to install things more properly (and then chmod the whole damned thing because nobody's ever taught the thing how to look for read-only and user config files, or figure out where its resources are located) and also to install into a DESTDIR if set for packaging.  I'd be just as happy to provide debian/* and someone else give us a linapple.spec file right in the repo when we're done.

Merge first.

Hm, y'know what I might do, if SDL's license still permits it (and I believe it will) is pull SDL 2's XDG compatibility function into a file with a conditional directive.  Whenever we get around to porting this sucker to SDL 2, it'll just not be compiled in.

(Not to self: Investigate [SDLCL](https://github.com/MrAlert/sdlcl) which is meant for old non-free Linux games via LD_PRELOAD, bet if you're willing to mass rename a few things you can turn it into a compile-time shim for SDL2 so we can port things as we look at them…  Post merge.)

**Comment by cpressey** [comment]
@iKarith I either did not notice that commit, or if I did I disregarded it as Mac-specific (in my fork I wasn't concerned about supporting that architecture.)

I don't have a lot of bandwidth to contribute to this, but I'm happy to try to help get any desirable changes out of my fork and into this, before it undergoes source layout reorganization that will make that harder.

Looking at it now, I think those desirable changes might only be:

https://github.com/catseye/linapple/pull/3/files - Set resource dir at build time. I understand this fixes "the 777 problem", but as a tangential note, making it safer when installed globally wasn't really my goal: my goal was to make it so that doesn't have to be installed at all (but could still be started from any directory in the filesystem.)  (And I like it much better when I can run software without installing it at all.)

https://github.com/catseye/linapple/pull/4/files -  Read-only linapple.conf.  This makes sense to me in an environment like Unix where you're expected to edit the configuation file in a text editor, instead of with a GUI within the emulator.  Partly I made this change also because otherwise, linapple writes out a zero-length `linapple.conf` every time you run it (because it opens the file in append mode.)  (And I found that annoying.)

(I mention these things here because these changes are at least partly philosophy, and my philosophy might not be everyone's philosophy.)

**Comment by dabonetn** [comment]
Ok, finally getting time to read some more of this.. 
So what the goal?

My was based on 2.0b and added all the command line options of catseye except for the log and directsound.
It also includes all the needed resource files into the executable, no more loading of external bmps using the sdl-image lib, so no need of a resource directory.
It checks for linapple.conf and Master.dsk in the ~/.linapple first 
and then checks the launch directory before writing that zero byte linapple.conf.
(I hadn't really noticed that until you pointed it out.)

**Comment by iKarith** [comment]
Also @cpressey, your tree pretty much has those two things, macOS support, and the README changes documenting it.  Greg's tree will break macOS support, but I do intend both a port to SDL2 and to fix the macOS support.  They're likely to happen in that order.  I'll do the SDL2 stuff in a branch and invite some folks to test it before committing that into this tree.

**Comment by iKarith** [comment]
I'm ready when you are—I've just had RL interfering lately with last merges of Greg's tree (given that I want to reduce the chances of committing some stupid bug when I do…)

**Comment by maxolasersquad** [comment]
It is done.

**Comment by iKarith** [comment]
Thanks David!  Just finished merging Greg's changes—will open a PR for a few days to give folks a chance to look at it, @cpressey in particular. Sorry Chris the raw diff is gonna be obnoxious, you might want to just look at relevant files.

Hey @ghedger, if you try to pull this before I try pushing it to you, note b450477 and that your pkg directory isn't gonna update cleanly because you chown it.  See the README in that commit.  :)

I was once a Debian developer, til I took an arrow to the knee…  At some point I'll do up a debian/ dir and modify make package accordingly.
